apps/testing-modal/src/app/app.component.spec.ts

```diff
@@ -1,16 +1,64 @@
-import { render } from '@testing-library/angular';
+import { screen, render, fireEvent } from '@testing-library/angular';
+import { within } from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
 import { AppComponent } from './app.component';
 
 describe('AppComponent', () => {
-  test('error modal is displayed if you click on "Confirm" without inputing a name', async () => {
+  const expectedNameInvalidErrorMessage = 'Name is invalid !!';
+  const expectedNameRequiredErrorMessage = 'You must enter a name first!!';
+
+  async function writeNameIntoControl(name: string) {
+    const control = screen.getByRole('textbox', { name: /name/i });
+    await userEvent.type(control, name);
+  }
+
+  function clickConfirmControl() {
+    const control = screen.getByText('Confirm');
+    fireEvent.click(control);
+  }
+
+  async function clickCancelControlInModal() {
+    const control = within(await getModal()).getByText('Cancel');
+    fireEvent.click(control);
+  }
+
+  async function clickConfirmControlInModal() {
+    const control = within(await getModal()).getByText('Confirmation');
+    fireEvent.click(control);
+  }
+
+  async function getModal(): Promise<HTMLElement> {
+    return screen.getByRole('dialog');
+  }
+
+  beforeEach(async () => {
     await render(AppComponent);
   });
 
+  test('error modal is displayed if you click on "Confirm" without inputing a name', async () => {
+    clickConfirmControl();
+
+    expect(
+      screen.getByText(
+        (content, element) =>
+          element?.textContent?.trim() === expectedNameRequiredErrorMessage
+      )
```

Comment:
I simply did that: 

```
  const errorText = await screen.getByText(errorText);
  expect(errorText).toBeInTheDocument();
```

Reply:
That doesn't work for me because of the `span` around the word "name" (I guess).

> TestingLibraryElementError: Unable to find an element with the text: You must enter a name first!!. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.

This is why I came up with the solution I implemented in this PR.

Reply:
that's weird 🧐